services/outputhost/outputhost.go

```diff
@@ -507,22 +507,16 @@ func (h *OutputHost) ReceiveMessageBatch(ctx thrift.Context, request *cherami.Re
 	moreResultTimer := common.NewTimer(timeoutTime.Sub(time.Now()))
 	defer moreResultTimer.Stop()
 MORERESULTLOOP:
-	for remaining := count - 1; remaining > 0; {
+	for remaining := count - 1; remaining > 0; remaining-- {
 		select {
-		case msg := <-cgCache.msgsRedeliveryCh:
-			deliver(msg, cgCache.msgCacheRedeliveredCh)
-			remaining--
-		case msg := <-cgCache.msgsCh:
-			deliver(msg, cgCache.msgCacheCh)
-			remaining--
+		case <-moreResultTimer.C:
+			break MORERESULTLOOP
 		default:
```

Comment:
which 'default'? Line 518?
I think that's still needed. So in each loop we first check whether it's timed out. Only if not time out(default case), we check both message availability and the timer.